test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1588 @@
+		for start := time.Now(); time.Since(start) < PollTimeout; time.Sleep(PollInterval) {
+			pod, err := podClient.Get(context.TODO(), pod.Name, metav1.GetOptions{})
+			if err != nil {
+				return nil, err
+			}
+			if !verifyPodAllocations(pod, expectedContainers, false) {
+				continue
+			}
+			return pod, nil
+		}
+		return nil, fmt.Errorf("timed out waiting for pod resource allocation values to match expected")
+	}
+	waitContainerCgroupValuesEqualsExpected := func() error {
+		for start := time.Now(); time.Since(start) < PollTimeout; time.Sleep(PollInterval) {
+			if !verifyPodContainersCgroupValues(ctx, f, patchedPod, expectedContainers, false) {
+				continue
+			}
+			return nil
+		}
+		return fmt.Errorf("timed out waiting for container cgroup values to match expected")
+	}
+	waitPodStatusResourcesEqualSpecResources := func() (*v1.Pod, error) {
+		for start := time.Now(); time.Since(start) < PollTimeout; time.Sleep(PollInterval) {
+			pod, err := podClient.Get(context.TODO(), pod.Name, metav1.GetOptions{})
+			if err != nil {
+				return nil, err
+			}
+			differs := false
+			for idx, c := range pod.Spec.Containers {
+				if !cmp.Equal(c.Resources, *pod.Status.ContainerStatuses[idx].Resources) {
+					differs = true
+					break
+				}
+			}
+			if differs {
+				continue
+			}
+			return pod, nil
+		}
+		return nil, fmt.Errorf("timed out waiting for pod spec resources to match pod status resources")
+	}
+	rsErr := waitForContainerRestart()
+	framework.ExpectNoError(rsErr, "failed waiting for expected container restart")
+	// Wait for pod resource allocations to equal expected values after resize
+	resizedPod, aErr := waitPodAllocationsEqualsExpected()
+	framework.ExpectNoError(aErr, "failed to verify pod resource allocation values equals expected values")
+	// TODO(vinaykul,InPlacePodVerticalScaling): Remove this check once base-OS updates to containerd>=1.6.9
+	//                containerd needs to add CRI support before Beta (See Node KEP #2273)
+	if !isInPlaceResizeSupportedByRuntime(c, pod.Spec.NodeName) {
+		// Wait for PodSpec container resources to equal PodStatus container resources indicating resize is complete
+		rPod, rErr := waitPodStatusResourcesEqualSpecResources()
+		framework.ExpectNoError(rErr, "failed to verify pod spec resources equals pod status resources")
+
+		ginkgo.By("verifying pod status after resize")
+		verifyPodStatusResources(rPod, expectedContainers)
+	} else if !framework.NodeOSDistroIs("windows") {
+		// Wait for container cgroup values to equal expected cgroup values after resize
+		// only for containerd versions before 1.6.9
+		cErr := waitContainerCgroupValuesEqualsExpected()
+		framework.ExpectNoError(cErr, "failed to verify container cgroup values equals expected values")
+	}
+	return resizedPod
+}
+
+func doPodResizeTests() {
+	f := framework.NewDefaultFramework("pod-resize-test")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
```

Comment:
nit: is this necessary? Looks like you're running restricted pods?

Reply:
Done